### PR TITLE
HDDS-8331. Add rocksdb option template file and docs

### PR DESCRIPTION
**Pull request by whbing**
## What changes were proposed in this pull request?

Add a template file and description to clarify how to customize the rocksdb options.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-8331

### CODE REVIEW

File: hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/DBConfigFromFile.java [hunk 1]
```diff
@@ -120,7 +120,7 @@ public static ManagedDBOptions readFromFile(String dbFileName,
     Preconditions.checkNotNull(cfDescs);
     Preconditions.checkArgument(cfDescs.size() > 0);
 
-    //TODO: Add Documentation on how to support RocksDB Mem Env.
+    // Documentation on how to support RocksDB Mem Env: see HDDS-8331.
```
Comment (kaijchen) [inline hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/DBConfigFromFile.java:123]: Maybe direct link is better?

```suggestion
    // Documentation on how to support RocksDB Mem Env:
    // see hadoop-hdds/docs/content/recipe/RocksdbOptions.md
```

File: hadoop-ozone/dist/src/shell/conf/db.ini.template [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+   compaction_readahead_size=4194304
+   bytes_per_sync=1048576
+   max_background_compactions=4
+   max_background_jobs=48
+   create_missing_column_families=true
+   create_if_missing=true
+
+[CFOptions "default"]
+
+[TableOptions/BlockBasedTable "default"]
```
Comment (kaijchen) [inline hadoop-ozone/dist/src/shell/conf/db.ini.template:25]: Add a newline before EOF.

File: hadoop-hdds/docs/content/recipe/RocksdbOptions.zh.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+   max_background_compactions=4
+   max_background_jobs=48
+   create_missing_column_families=true
+   create_if_missing=true
+
+[CFOptions "default"]
+
+[TableOptions/BlockBasedTable "default"]
+```
+重启OM后，查看OM的rocksdb目录（`${DBDATA}/om.db`）下`OPTIONS-xxxxxx`内容，`max_background_flushes=4`表明配置生效。
```
Comment (kaijchen) [inline hadoop-hdds/docs/content/recipe/RocksdbOptions.zh.md:55]: Please add a space between Chinese characters and English characters.

For example:
```
将对应的配置文件 `file` 放在OZONE的配置目录 `$OZONE_DIR/etc/hadoop` 下，然后重启服务即可。以下列出了 OM/SCM/DN 默认加载的 rocksdb 配置文件名。
```

**Review by kaijchen — COMMENTED** [review]
Thanks @whbing for the patch. Mostly looks good to me.
Just left some minor improvements.

File: hadoop-ozone/dist/src/shell/conf/db.ini.template [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# This is a RocksDB option file.
```
Comment (kaijchen) [inline hadoop-ozone/dist/src/shell/conf/db.ini.template:1]: We should include a license header in this file to pass the RAT check.
Is this file copied from RocksDB?

Comment (whbing) [reply hadoop-ozone/dist/src/shell/conf/db.ini.template:1]: > We should include a license header in this file to pass the RAT check.
> Is this file copied from RocksDB?

Copied from https://github.com/facebook/rocksdb/blob/main/examples/rocksdb_option_file_example.ini, and added some description

**Comment by whbing** [comment]
@kaijchen  Thanks! add new commit.

File: hadoop-ozone/dist/src/shell/conf/db.ini.template (hunk 4 above)
Comment (kaijchen) [reply hadoop-ozone/dist/src/shell/conf/db.ini.template:18]: Maybe we should keep the original author, and add this file in LICENSE.
Cc @adoroszlai, what do you think?

For example 
https://github.com/apache/ozone/blob/57d66137fa54d566752f2787e853deecda9d194b/LICENSE.txt#L216-L217

**Comment by kerneltime** [comment]
cc @tanvipenumudy can you please take a look?

**Comment by whbing** [comment]
@kaijchen @kerneltime  Adding a template file seems controversial. Just adding docs without template file maybe better.  
I will remove the template file if possible. What do you think?

**Review by adoroszlai — COMMENTED** [review]
Thanks @whbing for the patch.  Documenting this is very useful.  I would prefer this change to be restricted to adding the documentation, but not the config template.  It would be very easy for developers to miss updating the template (if necessary) when upgrading RocksDB in Ozone.

**Pull request closed without merging**